src/main/java/com/xceptance/xlt/engine/htmlunit/okhttp3/OkHttp3WebConnection.java

```diff
@@ -0,0 +1,380 @@
+
+    private static final EasyHostnameVerifier INSECURE_HOSTNAME_VERIFIER = new EasyHostnameVerifier();
+
+    private static final EasyX509TrustManager INSECURE_TRUST_MANAGER = new EasyX509TrustManager(null);
+
+    private static final SSLSocketFactory INSECURE_SSL_SOCKET_FACTORY = createInsecureSslSocketFactory();
+
+    private final AuthenticationCache authenticationCache;
+
+    private final ConnectionPool connectionPool;
+
+    private final DnsImpl dns;
+
+    private final List<Protocol> protocols;
+
+    /**
+     * Constructor.
+     *
+     * @param webClient
+     *            the owning web client
+     * @param http2Enabled
+     *            whether or not HTTP/2 is enabled at all
+     */
+    public OkHttp3WebConnection(final WebClient webClient, final boolean http2Enabled)
+    {
+        super(webClient);
+
+        authenticationCache = new AuthenticationCache();
+        connectionPool = new ConnectionPool(6, 60, TimeUnit.SECONDS);
+        dns = new DnsImpl(new XltDnsResolver());
+        protocols = http2Enabled ? HTTP_2_AND_1_1 : HTTP_1_1_ONLY;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected OkHttpClient createHttpClient(final WebClient webClient, final WebRequest webRequest) throws Exception
+    {
+        final WebClientOptions webClientOptions = webClient.getOptions();
+        final Authenticator authenticator = new AuthenticatorImpl(webClient.getCredentialsProvider());
+
+        final Builder httpClientBuilder = new OkHttpClient.Builder();
+
+        // basic settings and components
+        httpClientBuilder.authenticator(authenticator);
+        // httpClientBuilder.cache(null);
+        httpClientBuilder.connectionPool(connectionPool);
+        // httpClientBuilder.connectionSpecs(createConnectionSpecs(webClientOptions));
+        httpClientBuilder.cookieJar(new CookieJarImpl(webClient.getCookieManager()));
+        httpClientBuilder.dns(dns);
+        httpClientBuilder.followRedirects(false);
+        httpClientBuilder.followSslRedirects(false);
+        // httpClientBuilder.pingInterval(null);
+        httpClientBuilder.protocols(protocols);
+        httpClientBuilder.retryOnConnectionFailure(true);
+
+        // proxy
+        final String proxyHost = webRequest.getProxyHost();
+        if (proxyHost != null)
+        {
+            final InetSocketAddress proxyHostAddress = new InetSocketAddress(proxyHost, webRequest.getProxyPort());
+            final Proxy proxy = new Proxy(webRequest.isSocksProxy() ? Proxy.Type.SOCKS : Proxy.Type.HTTP, proxyHostAddress);
+            httpClientBuilder.proxy(proxy);
+            httpClientBuilder.proxyAuthenticator(authenticator);
+        }
+
+        // timeouts
+        int timeout = webRequest.getTimeout();
+        if (timeout < 0)
+        {
+            timeout = webClientOptions.getTimeout();
+        }
+
+        final Duration timeoutMS = Duration.ofMillis(timeout);
+        // httpClientBuilder.callTimeout(timeoutMS);
```

Comment:
Remove the configuration statements that have been commented out?

Reply:
Done.